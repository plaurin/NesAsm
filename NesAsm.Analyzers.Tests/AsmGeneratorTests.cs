using ApprovalTests.Reporters;

namespace NesAsm.Analyzers.Tests;

[UseReporter(typeof(VisualStudioReporter))]
public class AsmGeneratorTests
{
    [Fact]
    public void TestExampleScript()
    {
        var generator = new AsmGenerator();
        var content = File.ReadAllText($@"..\..\..\..\NesAsm.Example\HiloWorld.cs");

        GeneratorTestUtilities.TestGenerator(generator, content);
    }

    [Fact]
    public void TestSimpleScript()
    {
        TestScript("SimpleScript.cs");
    }

    [Fact]
    public void TestMultiProcScript()
    {
        TestScript("MultiProcScript.cs");
    }

    [Fact]
    public void TestJumpSubroutineScript()
    {
        TestScript("JumpSubroutineScript.cs");
    }

    [Fact]
    public void TestCharsScript()
    {
        TestScript("CharsScript.cs");
    }

    [Fact]
    public void TestBranchScript()
    {
        TestScript("BranchScript.cs");
    }

    [Fact]
    public void TestParametersScript()
    {
        TestScript("ParametersScript.cs");
    }

    [Fact]
    public void TestSubroutineResultScript()
    {
        TestScript("SubroutineResultScript.cs");
    }

    [Fact]
    public void TestSubroutineResultsScript()
    {
        TestScript("SubroutineResultsScript.cs");
    }

    [Fact]
    public void TestCallingOtherScript()
    {
        TestScript("CallingOtherScript.cs");
    }

    [Fact]
    public void TestDataScript()
    {
        TestScript("DataScript.cs");
    }

    [Fact]
    public void TestAllInstructionsScript()
    {
        TestScript("AllInstructionsScript.cs");
    }

    private static void TestScript(string filename)
    {
        var generator = new AsmGenerator();
        var content = File.ReadAllText($@"..\..\..\TestFiles\{filename}");

        GeneratorTestUtilities.TestGenerator(generator, content);
    }
}
