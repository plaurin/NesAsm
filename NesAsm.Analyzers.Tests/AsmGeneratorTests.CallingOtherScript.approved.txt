- Main diagnostics:
- SyntaxTrees Count: 2
- Output diagnostics:
The type or namespace name 'MultiProcScript' could not be found (are you missing a using directive or an assembly reference?)
- RunResult.GeneratedTrees Count: 1
- RunResults diagnostics:
- RunResult[0].Generator: Microsoft.CodeAnalysis.IncrementalGeneratorWrapper
- RunResult[0] diagnostics:
- RunResult[0].GeneratedSources Count: 1
- RunResult[0].Exception: 
- RunResult[0].GeneratedSources[0].HintName: Asm.CallingOtherScript.cs
- RunResult[0].GeneratedSources[0].SourceText:
/* Generator Asm code in file Asm.CallingOtherScript.s
.segment "CODE"

.proc main
  jsr procB
  jsr myProcC

  rts
.endproc

.proc myProcC
  lda #30

  rts
.endproc


*/
